File: web/nexus-ui/src/contracts/requests/change-password.request.ts
export interface ChangePasswordRequest {
    userId: string | null;
    verifier: string;
    clientSalt: string; 
    encryptedDek: string;
    cryptoVersion: number; 
    srpVersion: number;
    encryptedVerifierWrapKey: string; 
    keyWrapVersion: number; 
    asymmetricKeyId: string;
}

File: backend/src/shared/contracts/Shared.Contracts/UserManagement/Requests/ChangePasswordRequest.cs
namespace Shared.Contracts.UserManagement.Requests
{
    public sealed record ChangePasswordRequest(
        string? UserId,
        // Srp
        string EncryptedVerifier, 
        string SrpSalt, 
        int SrpVersion, 
        string EncryptedVerifierWrapKey, 
        int KeyWrapVersion, 
        string AsymmetricKeyId,
        // Dek
        string EncryptedDek,
        string DekSalt,
        int CryptoVersion);
}